common/unit_test/Test_Padding.cpp

```diff
@@ -43,434 +65,535 @@ void test_reshape1D_1DView(bool is_C2R) {
-          x, shape_type<1>{n_new}, axes_type<1>{axis0}, is_C2R);
+      std::size_t n_new   = static_cast<std::size_t>(x_in.extent(axis0) + i0);
+      ref_shape.at(axis0) = get_c2r_shape(n_new, is_C2R);
+
+      auto modified_shape = KokkosFFT::Impl::get_modified_shape(
+          x_in, x_out, shape_type<1>{n_new}, axes_type<1>{axis0});
 
       EXPECT_TRUE(modified_shape == ref_shape);
     }
   }
 }
 
-void test_reshape1D_5DView(bool is_C2R) {
+template <typename T>
+void test_reshape1D_5DView() {
   const int n0 = 5, n1 = 11, n2 = 10, n3 = 8, n4 = 3;
-  View5D<double> x("x", n0, n1, n2, n3, n4);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+  View5D<T> x_out("x_out", n0, n1, n2, n3, n4);
 
   constexpr int DIM = 5;
   shape_type<5> default_shape({n0, n1, n2, n3, n4});
 
   for (int axis0 = 0; axis0 < DIM; axis0++) {
+    shape_type<5> in_shape         = default_shape;
+    in_shape.at(axis0)             = get_c2r_shape(x_out.extent(axis0), is_C2R);
+    auto [_n0, _n1, _n2, _n3, _n4] = in_shape;
+    View5D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2, _n3, _n4);
     for (int i0 = -1; i0 <= 1; i0++) {
       shape_type<5> ref_shape = default_shape;
-      std::size_t n_new       = static_cast<std::size_t>(x.extent(axis0) + i0);
-      ref_shape.at(axis0)     = get_c2r_shape(n_new, is_C2R);
-      auto modified_shape     = KokkosFFT::Impl::get_modified_shape(
-          x, shape_type<1>{n_new}, axes_type<1>{axis0}, is_C2R);
+      std::size_t n_new   = static_cast<std::size_t>(x_in.extent(axis0) + i0);
+      ref_shape.at(axis0) = get_c2r_shape(n_new, is_C2R);
+
+      auto modified_shape = KokkosFFT::Impl::get_modified_shape(
+          x_in, x_out, shape_type<1>{n_new}, axes_type<1>{axis0});
 
       EXPECT_TRUE(modified_shape == ref_shape);
     }
   }
 }
 
-void test_reshape1D_6DView(bool is_C2R) {
+template <typename T>
+void test_reshape1D_6DView() {
   const int n0 = 5, n1 = 11, n2 = 10, n3 = 8, n4 = 3, n5 = 4;
-  View6D<double> x("x", n0, n1, n2, n3, n4, n5);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+  View6D<T> x_out("x_out", n0, n1, n2, n3, n4, n5);
 
   constexpr int DIM = 6;
   shape_type<6> default_shape({n0, n1, n2, n3, n4, n5});
-
   for (int axis0 = 0; axis0 < DIM; axis0++) {
+    shape_type<6> in_shape = default_shape;
+    in_shape.at(axis0)     = get_c2r_shape(x_out.extent(axis0), is_C2R);
+    auto [_n0, _n1, _n2, _n3, _n4, _n5] = in_shape;
+    View6D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2, _n3, _n4, _n5);
     for (int i0 = -1; i0 <= 1; i0++) {
       shape_type<6> ref_shape = default_shape;
-      std::size_t n_new       = static_cast<std::size_t>(x.extent(axis0) + i0);
-      ref_shape.at(axis0)     = get_c2r_shape(n_new, is_C2R);
-      auto modified_shape     = KokkosFFT::Impl::get_modified_shape(
-          x, shape_type<1>{n_new}, axes_type<1>{axis0}, is_C2R);
+      std::size_t n_new   = static_cast<std::size_t>(x_in.extent(axis0) + i0);
```

Comment:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.